fix(icons): Icon path must be relative to plugin registry root folder

vitaliy-guliy opened the pull request:
Signed-off-by: Vitaliy Gulyy <vgulyy@redhat.com>

<!-- Please review the following before submitting a PR:
Che's Contributing Guide: https://github.com/eclipse/che/blob/master/CONTRIBUTING.md
Pull Request Policy: https://github.com/eclipse/che/wiki/Development-Workflow#pull-requests

-->

### What does this PR do?
Applies changes from this PR https://github.com/eclipse-che/che-plugin-registry/pull/976

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-1840

Upstream issue
https://github.com/eclipse/che/issues/19771

svor commented:
@ericwill shouldn't `/images/eclipse-che-logo.png` be replaced with something else which is more relevant to crw?

vitaliy-guliy commented:
> @ericwill shouldn't `/images/eclipse-che-logo.png` be replaced with something else which is more relevant to crw?

There is `default.svg`, fully transparent image.

vitaliy-guliy commented:
I tried apply this icon https://raw.githubusercontent.com/redhat-developer/codeready-workspaces/crw-2-rhel-8/product/branding/CodeReady.png

and got

![Screenshot from 2021-06-03 16-45-37](https://user-images.githubusercontent.com/1655894/120655540-722d4600-c48b-11eb-9733-85d36d572a49.png)

![Screenshot from 2021-06-03 16-45-48](https://user-images.githubusercontent.com/1655894/120655545-75283680-c48b-11eb-9bd2-29130aa50b88.png)

So, applying it

nickboldt commented on dependencies/che-plugin-registry/tools/build/tests/meta-yaml/meta-yaml-writer.spec.ts:
```diff
@@ -417,7 +417,7 @@ spec: {}
       title: 'minimal-endpoint',
       description: 'minimal-endpoint',
-      icon: '/v3/images/eclipse-che-logo.png',
+      icon: '/v3/images/CodeReady.png',
```
why is this in /v3/images but in other places you have only /images/ folder?

vitaliy-guliy replied:
fixed

nickboldt commented on dependencies/che-plugin-registry/tools/build/tests/che-theia-plugin/che-theia-plugins-yaml-writer.spec.ts:
```diff
@@ -231,7 +231,7 @@ metadata:
     - Programming Languages
   repository: 'http://fake-repository'
-  icon: /v3/images/eclipse-che-logo.png
+  icon: /v3/images/CodeReady.png
```
in other places you have /images/ instead of /v3/images/ ... is that correct?

nickboldt replied:
> could we use only lowercase icons name please ?

And if you're renaming it ... please call it codeready-workspaces-logo.png instead of codeready... because (say it with me everyone!)...

* CodeReady is a brand which includes several products!
* CodeReady Workspaces is the single product which uses this icon/logo.

vitaliy-guliy replied:
Then it should be renamed here as well https://github.com/redhat-developer/codeready-workspaces/tree/crw-2-rhel-8/product/branding

vitaliy-guliy replied:
> in other places you have /images/ instead of /v3/images/ ... is that correct?

It's a test, but I will fix it to be less confusing.

nickboldt replied:
> Then it should be renamed here as well https://github.com/redhat-developer/codeready-workspaces/tree/crw-2-rhel-8/product/branding

Damn! you got me :/

vitaliy-guliy replied:
> in other places you have /images/ instead of /v3/images/ ... is that correct?

used transparent default.png  and removed /v3

nickboldt commented on dependencies/che-plugin-registry/tools/build/src/meta-yaml/meta-yaml-writer.ts:
```diff
@@ -31,7 +31,12 @@ export class MetaYamlWriter {
+  //
+  // It must work also for single root deployments
+  //    https://che-eclipse-che.apps-crc.testing/plugin-registry/v3
+  public static readonly DEFAULT_ICON = '/images/CodeReady.png';
```
in other places you have /v3/images/ instead of /images/ ... which is correct?

vitaliy-guliy replied:
without /v3 is correct

ericwill reviewed (commented):
Can you also please a line to the Dockerfile that copies the CRW icon into the `v3/images` folder? Similar to this: https://github.com/eclipse-che/che-plugin-registry/blob/master/build/dockerfiles/Dockerfile#L26

but after this line instead: https://github.com/redhat-developer/codeready-workspaces/blob/crw-2-rhel-8/dependencies/che-plugin-registry/build/dockerfiles/Dockerfile#L87

nickboldt commented:
Note: Officially, we should be involving Brand in any new icon changes. While the CRW logo looks ok on grey, it was not designed for that and Brand might want to keyline the logo in black or white so it pops better on the light grey background. We've also been told that the product should not show the logo more than 2 times on the screen ... adding it into the plugin install view waters it down and makes it less effective. 

So... to be really safe and compliant with Brand rules, we should stick with the 1px invisible png, and only display plugin icons for 3rd party stuff. Anything that's a CRW plugin should simply have no logo.

ericwill commented:
> Note: Officially, we should be involving Brand in any new icon changes. While the CRW logo looks ok on grey, it was not designed for that and Brand might want to keyline the logo in black or white so it pops better on the light grey background. We've also been told that the product should not show the logo more than 2 times on the screen ... adding it into the plugin install view waters it down and makes it less effective.
> 
> So... to be really safe and compliant with Brand rules, we should stick with the 1px invisible png, and only display plugin icons for 3rd party stuff. Anything that's a CRW plugin should simply have no logo.
> 
> Discussions re: logos in plugin view:
> 
>     * https://issues.redhat.com/browse/CRW-1385
> 
>     * https://issues.redhat.com/browse/CRW-675
> 
>     * https://issues.redhat.com/browse/CRW-561
> 
> 
> cc: @bmorley

+1 to just using a transparent icon instead of the CRW one.

vitaliy-guliy commented:
> Note: Officially, we should be involving Brand in any new icon changes. While the CRW logo looks ok on grey, it was not designed for that and Brand might want to keyline the logo in black or white so it pops better on the light grey background. We've also been told that the product should not show the logo more than 2 times on the screen ... adding it into the plugin install view waters it down and makes it less effective.
> 
> So... to be really safe and compliant with Brand rules, we should stick with the 1px invisible png, and only display plugin icons for 3rd party stuff. Anything that's a CRW plugin should simply have no logo.
> 
> Discussions re: logos in plugin view:
> 
> * https://issues.redhat.com/browse/CRW-1385
> * https://issues.redhat.com/browse/CRW-675
> * https://issues.redhat.com/browse/CRW-561
> 
> cc: @bmorley

OK, adding a transparent image then.

nickboldt commented:
Other reason for transparent image... in the screenshots above I see broadcomm and buildah plugins (both not included in CRW) using the CRW logo. That's definitely wrong -- if the plugin doesn't provide an icon, we shouldn't put the product icon on it.

vitaliy-guliy commented:
> Other reason for transparent image... in the screenshots above I see broadcomm and buildah plugins (both not included in CRW) using the CRW logo. That's definitely wrong -- if the plugin doesn't provide an icon, we shouldn't put the product icon on it.

Yes, I just changed uri to the icon on my local Che deployment. Just to take the screenshots.
I've already added transparent 1x1 px image and used it by default.

nickboldt reviewed (approved):
+1 LGTM. Please remember to cherry pick this to the 2.9 branch too if we want it in 2.9.

svor reviewed (approved):
@vitaliy-guliy Please resolve conflicts and do cherry-pick into 2.9 branch

vitaliy-guliy commented:
@svor @ericwill this PR is updated, but seems something wrong with Build PR check.
Could you review the PR again?

It needs to be decided what shall we do with those both PR checks.

ericwill commented:
> @svor @ericwill this PR is updated, but seems something wrong with Build PR check.
> Could you review the PR again?
> 
> It needs to be decided what shall we do with those both PR checks.

PR check will fail because the PR is opened from your fork, not the origin

svor commented:
@ericwill I think most of PRs will be created from forks. As I know for upstream it's not a problem, do we have some restrictions to build PR from the fork?

nickboldt commented:
> do we have some restrictions to build PR from the fork?

If you want PR check tests (that require the use of secrets) to pass, then you can't run those PR checks from forks.

nickboldt commented:
OK so the options here:
* resubmit the PR using a branch against the origin, instead of your fork
* ignore the failing tests and hope for the best.

Also need same fix for the backport PR, #517

vitaliy-guliy commented:
> OK so the options here:
> 
> * resubmit the PR using a branch against the origin, instead of your fork
> * ignore the failing tests and hope for the best.
> 
> Also need same fix for the backport PR, #517

No problems here, I will resubmit PR when I have permissions on create branches.

vitaliy-guliy commented:
Superseded by https://github.com/redhat-developer/codeready-workspaces/pull/518
